crypto-app/src/pages/Coins/Coins.js

```diff
@@ -0,0 +1,131 @@
+import React from 'react';
+import { CoinInstance } from "components";
+import InfiniteScroll from 'react-infinite-scroll-component';
+import { TableContainer, TableHeader, Table, TableRow } from './styles';
+import axios from 'axios';
+
+class Coins extends React.Component {
+  state = {
+    coins:[],
+    page: 1,
+    isLoading: false,
+    hasError: false,
+    errMess: ""
+  }
+  getCoins = async () => {
+    try {
+      this.setState({
+        isLoading: true
+      })
+      const { data } = await axios(
+        `https://api.coingecko.com/api/v3/coins/markets?vs_currency=usd&order=market_cap_desc&per_page=50&page=1&sparkline=true&price_change_percentage=1h%2C24h%2C7d`
+      );
+      this.setState({
+        coins: data,
+        isLoading: false,
+      });
+    } catch(error) {
+      this.setState({
+        isLoading: false,
+        hasError: true,
+        errMess: "There was a problem getting coin list!"
+      })
+    }
+  }
+  getMoreCoins = async () => {
+    try {
+      this.setState({
+        page: this.state.page + 1,
+        isLoading: true,
+      });
+      const  { data }  = await axios(
+        `https://api.coingecko.com/api/v3/coins/markets?vs_currency=usd&order=market_cap_desc&per_page=50&page=${this.state.page}&sparkline=true&price_change_percentage=1h%2C24h%2C7d`
```

Comment:
this.state.page in your query will be the previous page, not the one you setup in the beginning FYI